feat: Company list selector

peterhudec opened the pull request:
# Description of Change

Added a dropdown to the dashboard to allow for selecting company lists.

## Screenshots
<img width="916" alt="Screen Shot 2019-10-04 at 11 31 15 AM" src="https://user-images.githubusercontent.com/2333157/66201157-89a2a300-e69a-11e9-9994-914ad63e371b.png">
<img width="917" alt="Screen Shot 2019-10-04 at 11 30 39 AM" src="https://user-images.githubusercontent.com/2333157/66201158-89a2a300-e69a-11e9-95ca-aadca913227d.png">
<img width="917" alt="Screen Shot 2019-10-04 at 11 31 05 AM" src="https://user-images.githubusercontent.com/2333157/66201160-89a2a300-e69a-11e9-93e9-fa2bd91f5cf1.png">
<img width="917" alt="Screen Shot 2019-10-04 at 11 30 54 AM" src="https://user-images.githubusercontent.com/2333157/66201162-89a2a300-e69a-11e9-9d2c-932cf5a2dcc4.png">

ian-leggett commented on src/dashboard/my-companies/ListSelector.jsx:
```diff
@@ -0,0 +1,80 @@
+  margin-right: 10px;
+`
+
+const StyledSelect = styled(Select)`
```
I believe  every time we use styled components we are name spacing it like `StyledDiv`.

ian-leggett commented on src/dashboard/my-companies/ListSelector.jsx:
```diff
@@ -0,0 +1,80 @@
+const StyledSelect = styled(Select)`
+  display: inline-block;
+  margin-right: 10px;
+  // The underlying select has for some reason witdh 50%
```
I think we should remove this comment

ian-leggett commented on src/dashboard/my-companies/ListSelector.jsx:
```diff
@@ -0,0 +1,80 @@
+                </>
+              )
+            }
+            <Link href="#foo">
```
This will need to go and be passed as a prop

ian-leggett commented on src/dashboard/my-companies/__tests__/ListSelector.test.jsx:
```diff
@@ -0,0 +1,77 @@
+        <ListSelector />
+      </useMyCompaniesContext.Provider>,
+    )
+    expect(wrapper.text()).toBe('My companies listFooEdit lists')
```
Should this be split into a test for the h2,h3 and anchor tag

peterhudec replied:
I'm testing on text only intentionally. Testing the markup is more of a _white box_ testing. The markup is implementation detail. We could change it without affecting the component's functionality, but with breaking the tests. The specs don't say anything about markup so we should not test it.

ian-leggett commented on src/dashboard/my-companies/useMyCompaniesContext.jsx:
```diff
@@ -1,19 +1,13 @@
-    recent: orderBy(companies, [c => c.latestInteraction.date || ''], ['desc']),
-    'least-recent': orderBy(companies, [c => c.latestInteraction.date || ''], ['asc']),
-    alphabetical: orderBy(companies, [c => c.company.name], ['asc']),
+    recent: _.orderBy(companies, [c => c.latestInteraction.date || ''], ['desc']),
```
Not a big one but we tend to deconstruct what we need from lodash rather than use the whole library with `_`

rafenden commented on src/dashboard/my-companies/ListSelector.jsx:
```diff
@@ -0,0 +1,80 @@
+                  <RightSideItem as="span">
+                    View list
+                  </RightSideItem>
+                  <StyledSelect
```
I suggest passing `View list` as `label` prop, e.g,

```
<StyledSelect label="View list">
```

rafenden commented on src/dashboard/my-companies/ListSelector.jsx:
```diff
@@ -0,0 +1,80 @@
+              )
+            }
+            <Link href="#foo">
+              Edit lists
```
Should that be `Edit list` for a single list?

rafenden commented on src/dashboard/my-companies/ListSelector.jsx:
```diff
@@ -0,0 +1,80 @@
+      <Headline>
+        My companies list
+      </Headline>
+      {lists.length
```
Could be simplified to:
```
{lists.length && (
  <>
  ...
  </>
}
```

rafenden commented on src/dashboard/my-companies/ListSelector.jsx:
```diff
@@ -0,0 +1,80 @@
+            </Link>
+          </>
+        )
+        : null
```
Shall we display some text to the user in case when there are no lists added? i.e.
```
You didn't create any lists yet, <Link>add one now.
```

paulgain commented on src/dashboard/my-companies/MyCompaniesTable.jsx:
```diff
@@ -23,8 +26,27 @@ const StyledDateCell = styled(Table.Cell)(typography.font({ size: 14 }), {
+  const sort = {
+    recent: _.orderBy(companies, [c => c.latestInteraction.date || ''], ['desc']),
+    'least-recent': _.orderBy(companies, [c => c.latestInteraction.date || ''], ['asc']),
+    alphabetical: _.orderBy(companies, [c => c.company.name], ['asc']),
```
For the sake of consistency can we use Lodash like so:
```
import { orderBy } from 'lodash'
...
recent: orderBy(companies, [c => c.latestInteraction.date || ''], ['desc']),
...
```

paulgain commented:
@peterhudec just a friendly reminder about our code review guidelines.
https://github.com/uktrade/data-hub-frontend/blob/develop/docs/Code%20review%20guidelines.md

**Under PR hygiene** 
"PRs shouldn't normally add or change more than 400 lines of code at the time".

peterhudec commented:
Addressed all comments. Sorry for the size of the PR, next time I'll split it into multiple PRs. Not that sure about the `rebase -i` @debitan, as it contradicts the idea of small atomic commits.

peterhudec commented:
There's been a last minute change to the acceptance criteria: The "edit lists" link on the top right should say "delete this list" and it should expose a way to the client (frontend app) to open the _delete list_ page.
